Add ConfigCat provider to Feature Flag Data in RUM (#23489)
DataDog/documentation: content/en/real_user_monitoring/guide/setup-feature-flag-data-collection.md

{{% /tab %}}
{{< /tabs >}}

### ConfigCat integration

{{< tabs >}}
{{% tab "Browser" %}}

> Initialize the ConfigCat Javascript SDK with subscribing to the `flagEvaluated` event that will report feature flag evaluations to Datadog:

Reviewer: ```suggestion
Initialize the ConfigCat Javascript SDK with subscribing to the `flagEvaluated` event that reports feature flag evaluations to Datadog:
```

Author: I also applied `by` instead of `with` to ensure consist texts.

```javascript
  const configCatClient = configcat.getClient(
  '#YOUR-SDK-KEY#',
  configcat.PollingMode.AutoPoll,
  {
    setupHooks: (hooks) =>
      hooks.on('flagEvaluated', (details) => {
        datadogRum.addFeatureFlagEvaluation(details.key, details.value);
      })
  }
  );
```

> For more information about initializing the ConfigCat Javascript SDK, see [Documentation](https://configcat.com/docs/sdk-reference/js).

Reviewer: ```suggestion
For more information about initializing the ConfigCat Javascript SDK, see ConfigCat's [JavaScript SDK documentation][1].

[1]: https://configcat.com/docs/sdk-reference/js
```

{{% /tab %}}
{{% tab "iOS" %}}

> Initialize the ConfigCat Swift (iOS) SDK with subscribing to the `flagEvaluated` event that will report feature flag evaluations to Datadog:

Reviewer: ```suggestion
Initialize the ConfigCat Swift iOS SDK by subscribing to the `flagEvaluated` event that reports feature flag evaluations to Datadog:
```

```swift
  let client = ConfigCatClient.get(sdkKey: "#YOUR-SDK-KEY#") { options in
    options.hooks.addOnFlagEvaluated { details in
        RUMMonitor.shared().addFeatureFlagEvaluation(featureFlag: details.key, variation: details.value)
    }
  }
```

> For more information about initializing the ConfigCat Swift (iOS) SDK, see [Documentation](https://configcat.com/docs/sdk-reference/ios).

Reviewer: ```suggestion
For more information about initializing the ConfigCat Swift (iOS) SDK, see ConfigCat's[Swift iOS SDK documentation][1].

[1]: https://configcat.com/docs/sdk-reference/ios
```

{{% /tab %}}
{{% tab "Android" %}}

> Initialize the ConfigCat Android SDK with subscribing to the `flagEvaluated` event that will report feature flag evaluations to Datadog:

Reviewer: ```suggestion
Initialize the ConfigCat Android SDK by subscribing to the `flagEvaluated` event that reports feature flag evaluations to Datadog:
```

```java
  ConfigCatClient client = ConfigCatClient.get("#YOUR-SDK-KEY#", options -> {
    options.hooks().addOnFlagEvaluated(details -> {
        GlobalRumMonitor.get().addFeatureFlagEvaluation(details.key, details.value);
    });
  });
```

> For more information about initializing the ConfigCat Android SDK, see [Documentation](https://configcat.com/docs/sdk-reference/android).

Reviewer: ```suggestion
For more information about initializing the ConfigCat Android SDK, see ConfigCat's [Android SDK documentation][1].

[1]: https://configcat.com/docs/sdk-reference/android
```

{{% /tab %}}
{{% tab "Flutter" %}}

> Initialize the ConfigCat Dart SDK with subscribing to the `flagEvaluated` event that will report feature flag evaluations to Datadog:

Reviewer: ```suggestion
Initialize the ConfigCat Dart SDK by subscribing to the `flagEvaluated` event that reports feature flag evaluations to Datadog:
```

```dart
  final client = ConfigCatClient.get(
    sdkKey: '#YOUR-SDK-KEY#',
    options: ConfigCatOptions(
        pollingMode: PollingMode.autoPoll(),
        hooks: Hooks(
            onFlagEvaluated: (details) => {
              DatadogSdk.instance.rum?.addFeatureFlagEvaluation(details.key, details.value);
            }
        )
    )
  );
```

> For more information about initializing the ConfigCat Dart SDK, see [Documentation](https://configcat.com/docs/sdk-reference/dart).

Reviewer: ```suggestion
For more information about initializing the ConfigCat Dart (Flutter) SDK, see ConfigCat's [Dart SDK documentation][1].

[1]: https://configcat.com/docs/sdk-reference/dart
```

{{% /tab %}}

{{% tab "React Native" %}}

> Initialize the ConfigCat React SDK with subscribing to the `flagEvaluated` event that will report feature flag evaluations to Datadog:

Reviewer: ```suggestion
Initialize the ConfigCat React SDK by subscribing to the `flagEvaluated` event that reports feature flag evaluations to Datadog:
```

```typescript
<ConfigCatProvider
  sdkKey="YOUR_SDK_KEY"
  pollingMode={PollingMode.AutoPoll}
  options={{
    setupHooks: (hooks) =>
      hooks.on('flagEvaluated', (details) => {
        DdRum.addFeatureFlagEvaluation(details.key, details.value);
      }),
  }}
>
  ...
</ConfigCatProvider>
```

> For more information about initializing the ConfigCat Dart SDK, see [Documentation](https://configcat.com/docs/sdk-reference/react).

Reviewer: ```suggestion
For more information about initializing the ConfigCat React SDK, see ConfigCat's [React SDK documentation][1].

[1]: https://configcat.com/docs/sdk-reference/react
```

{{% /tab %}}
{{< /tabs >}}

### ConfigCat integration

{{< tabs >}}
{{% tab "Browser" %}}

Initialize the ConfigCat Javascript SDK by subscribing to the `flagEvaluated` event that reports feature flag evaluations to Datadog:

```javascript
  const configCatClient = configcat.getClient(
  '#YOUR-SDK-KEY#',
  configcat.PollingMode.AutoPoll,
  {
    setupHooks: (hooks) =>
      hooks.on('flagEvaluated', (details) => {
        datadogRum.addFeatureFlagEvaluation(details.key, details.value);
      })
  }
  );
```

For more information about initializing the ConfigCat Javascript SDK, see ConfigCat's [JavaScript SDK documentation][1].
	
[1]: https://configcat.com/docs/sdk-reference/js

{{% /tab %}}
{{% tab "iOS" %}}

Initialize the ConfigCat Swift iOS SDK by subscribing to the `flagEvaluated` event that reports feature flag evaluations to Datadog:

```swift
  let client = ConfigCatClient.get(sdkKey: "#YOUR-SDK-KEY#") { options in
    options.hooks.addOnFlagEvaluated { details in
        RUMMonitor.shared().addFeatureFlagEvaluation(featureFlag: details.key, variation: details.value)
    }
  }
```

> For more information about initializing the ConfigCat Swift (iOS) SDK, ConfigCat's[Swift iOS SDK documentation][1].

Reviewer: ```suggestion
For more information about initializing the ConfigCat Swift (iOS) SDK, see ConfigCat's[Swift iOS SDK documentation][1].
```
